kernel/src/loader/mod.rs

```diff
@@ -33,6 +34,34 @@ pub enum Error {
     SeekElfStart,
     SeekProgramHeader,
 }
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match *self {
+                Error::BigEndianElfOnLittle => "Cannot convert Big Endian ELF to Little Endian",
+                Error::CommandLineCopy => "Failed to copy the command line string to guest memory",
+                Error::CommandLineOverflow => "Command line string overflows guest memory",
+                Error::InvalidElfMagicNumber => "Invalid ELF magic number",
+                Error::InvalidEntryAddress => "Invalid entry address found in ELF header",
+                Error::InvalidProgramHeaderSize => "Invalid ELF program header size",
+                Error::InvalidProgramHeaderOffset => "Invalid ELF program header offset",
+                Error::InvalidProgramHeaderAddress => "Invalid ELF program header address",
+                Error::ReadElfHeader => "Failed to read ELF header from the input VM image",
+                Error::ReadKernelImage => "Failed to read VM image and write it to guest memory",
+                Error::ReadProgramHeader => "Failed to read ELF program headers",
+                Error::SeekKernelStart => "Failed to seek to file offsets in ELF program headers",
```

Comment:
`p_offset` stores the offset to the segment the program header refers to so I would suggest:
```suggestion
                Error::SeekKernelStart => "Failed to seek to file offset as pointed by the ELF program header",
```